crowbar_framework/app/models/dns_service.rb

```diff
@@ -49,6 +49,11 @@ def validate_proposal_after_save proposal
       validation_error("At least one nameserver or one node with the dns-server role must be specified.")
     end
 
+    if proposal["attributes"]["dns"]["enable_designate"]
+      rndckey = proposal["attributes"]["dns"]["designate_rndc_key"]
+      validation_error("designate_rndc_key cannot be empty with designate enabled") if rndckey.empty?
```

Comment:
I wonder if we want to just generate one? that would basically be a migration for the dns proposal..

Reply:
generated one during the migration. but I would still like to keep the check